### ISSUE TITLE
fatal: --local can only be used inside a git repository

### ISSUE THREAD

**Issue by jsoref** [actions/checkout#714]
https://github.com/actions/checkout/blob/ec3a7ce113134d7a93b817d10a8272cb61118579/src/git-command-manager.ts#L365

https://github.com/jsoref/examples-testing/runs/5404530182?check_suite_focus=true#step:4:20

```
Download action repository 'actions/checkout@v2' (SHA:ec3a7ce113134d7a93b817d10a8272cb61118579)
...
Run actions/checkout@v2
  with:
    repository: jsoref/examples-testing
    token: ***
    ssh-strict: true
    persist-credentials: true
    clean: true
    fetch-depth: 1
    lfs: false
    submodules: false
  env:
    PR_HEAD_SHA: 2f3e6fc9852caf53e728702a7adf0e871e2e1531
    result_code: 
Syncing repository: jsoref/examples-testing
Getting Git version info
/usr/bin/git config --local --get remote.origin.url
Error: fatal: --local can only be used inside a git repository
Deleting the contents of '/home/runner/work/examples-testing/examples-testing'
```

**Comment by stmcginnis** [comment]
I am seeing similar behavior with a job that does the checkout, runs a script, then tries to use `git update-index`. I get the error:

```
+ git update-index --assume-unchanged cmd/cli/plugin/pinniped-auth/asset/pinniped-v0.4.4
fatal: not a git repository (or any of the parent directories): .git
```

I thought at first this might be a side effect of the default shallow clone, so I tried adding:

```yaml
with:
   fetch-depth: 0
```

As indicated in the README to get the full history. But no difference in behavior.

**Comment by jsoref** [comment]
My guess is that it's from this:
https://github.com/actions/checkout/blob/ec3a7ce113134d7a93b817d10a8272cb61118579/src/git-directory-helper.ts#L28-L33

The boolean logic is giving me a headache.

**Comment by jsoref** [comment]
Ok, figured it out.

Here's the debugging showing what happened:
https://github.com/jsoref/examples-testing/runs/5497620370?check_suite_focus=true#step:4:1

> [Run jsoref/check-spelling@prerelease](https://github.com/jsoref/examples-testing/runs/5497620370?check_suite_focus=true#step:2:1)
  -- _this added a file into a `.git/` directory_

> Run actions/checkout@v3
[Getting Git version info](https://github.com/jsoref/examples-testing/runs/5497620370?check_suite_focus=true#step:5:15)
  Working directory is '/home/runner/work/examples-testing/examples-testing'
  /usr/bin/git version
  git version 2.35.1
/usr/bin/git config --local --get remote.origin.url
Error: fatal: --local can only be used inside a git repository

**Comment by Koushikrad** [comment]
@jsoref , am facing the same error in checkout actions. Is this fixed? Is there a recommended fix for this?

**Comment by jsoref** [comment]
@Koushikrad: before you run your checkout action, do a `find .`:
https://github.com/jsoref/examples-testing/blob/77f41bc266cf5fe28fdbddc68f83cfd57a48b4fe/.github/workflows/spelling.yml#L130-L131

You'll probably see that there's a file that's squatting in a mostly empty `.git` directory.

**Comment by koushik-radhakrishan** [comment]
@jsoref , yes when I added `find .` am seeing a lot of files being listed. They are mostly my external dependencies from my project. Can you kindly help what i should be doing about those files? and how they related to the fatal error?

**Comment by Koushikrad** [comment]
@jsoref , yes when I added `find .` am seeing a lot of files being listed. They are mostly my external dependencies from my project. Can you kindly help what i should be doing about those files? and how they related to the fatal error?

**Comment by jsoref** [comment]
It's almost certainly a sequencing order. You're probably doing something *before* the checkout step that you should be doing *after*.

Fwiw, from memory `fatal` is somewhat inaccurate, in that, I'm pretty sure it just shoots everything and moves on w/ life w/o actually killing your job...
